### PR TITLE
Feature refactor trapd hzn 636

### PR DESCRIPTION
**Pull request by cernerpradeep**
HZN-636
Add ability to fetch Trapd config via REST

Following changes have been done.
1. Cretated TrapdConfigurationResource.java file.
2. Configured Trapd resource in Rest service class ConfigRestService.
3. Created TrapdConfiguration.java file in jaxb and deleted the castor marshall/unmarshall
4. Created the junit TrapdConfigurationResourceIT.java. We are getting proper response.

Need help with:
What should we assert in response in TrapdConfigurationResourceIT.java.

### CODE REVIEW

File: opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/trapd/TrapdConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+ //- Imported classes and packages -/
+//---------------------------------/
+
+/**
+ * Top-level element for the trapd-configuration.xml
+ *  configuration file.
+ * 
+ * @version $Revision$ $Date$
+ */
+@XmlRootElement(name = "trapd-Configuration")
```
Comment (soleger) [inline opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/trapd/TrapdConfiguration.java]: Make this trapd-configuration (lowercase c).

File: opennms-config/src/main/java/org/opennms/netmgt/config/TrapdConfigFactory.java [hunk 2]
```diff
@@ -196,14 +196,14 @@ public synchronized boolean getNewSuspectOnTrap() {
...
+        for (SnmpV3User user : m_config.getSnmpv3UserCollection()) {
             snmpUsers.add(new SnmpV3User(
                     user.getEngineId(),
                     user.getSecurityName(),
                     user.getAuthProtocol(),
-                    user.getAuthPassphrase(),
-                    user.getPrivacyProtocol(),
-                    user.getPrivacyPassphrase()));
+                    user.getAuthPassPhrase(),
+                    user.getAuthProtocol(),
```
Comment (soleger) [inline opennms-config/src/main/java/org/opennms/netmgt/config/TrapdConfigFactory.java]: Something is wrong here, you should still call the getPrivacyProtocol() and getPrivacyPassphrase() methods here.

**Comment by soleger** [comment]
You need to copy over the Castor-generated Snmpv3User class in opennms-config-jaxb as well and add JAXB annotations to it. I think we still need two versions of this class, one at org.opennms.netmgt.config.trapd.Snmpv3User and the other at org.opennms.netmgt.snmp.SnmpV3User. We don't want to add JAXB annotations to org.opennms.netmgt.snmp.SnmpV3User because that would create extra dependencies in the core/snmp-api project.

**Comment by cernerpradeep** [comment]
Hi Seth, 

By mistake I did a git pull from main branch and got files that I did not commit. So I closed this PR and created one more.

**Pull request closed without merging**